src/core/bpf-firewall.c

```diff
@@ -651,31 +651,24 @@ int bpf_firewall_load_custom(Unit *u) {
 }
 
 static int attach_custom_bpf_progs(Unit *u, const char *path, int attach_type, Set **set, Set **set_installed) {
-        BPFProgram *prog;
+        _cleanup_(bpf_program_freep) BPFProgram *prog;
```

Comment:
Please also see a comment for `bpf_program_serialize_attachment_set`.

When `*set` is empty, `prog` remains uninitialized: it's undefined behavior.
Also `_cleanup_` introduces double-free, please leave it as `BPFProgram *prog;`